src/main/java/hudson/remoting/RemoteInvocationHandler.java

```diff
@@ -443,7 +460,37 @@ public void run() {
                     } catch (InterruptedException e) {
                         logger.log(Level.FINE, "Interrupted", e);
                     }
-                    if (System.nanoTime() > nextSweep) {
+                    if (System.nanoTime() - nextSweep > 0) {
+                        // these limits are just guesses, but if there is a lot of references being drained
+                        // per sweep then likely these references are causing GC pressure
+                        if (count > 2000) {
+                            if (retainOrigin) {
+                                logger.log(Level.INFO,
+                                        "Unexported {0} references since last sweep, if this is frequent consider "
+                                                + "setting system property hudson.remoting.RemoteInvocationHandler"
+                                                + ".Unexporter.retainOrigin=false to reduce memory pressure",
+                                        count
+                                );
+                            } else {
+                                logger.log(Level.INFO, "Unexported {0} references since last sweep", count);
+                            }
+                        } else if (count > 500) {
+                            if (retainOrigin) {
+                                logger.log(Level.FINE,
+                                        "Unexported {0} references since last sweep, if this is frequent consider "
+                                                + "setting system property hudson.remoting.RemoteInvocationHandler"
+                                                + ".Unexporter.retainOrigin=false to reduce memory pressure",
+                                        count
+                                );
+                            } else {
+                                logger.log(Level.FINE, "Unexported {0} references since last sweep", count);
```

Comment:
May be confusing, because people may understand is as a statistics for any logging level. Maybe "Unexported more than 500 references since last sweep (number = {0})"